ThunderCatsTests/UIColorTests.m

```diff
@@ -30,101 +30,70 @@
+    XCTAssertEqualWithAccuracy(red, 255.0/255.0, 0.05,  @"Expected red value of 1, but got %f", red);
+    XCTAssertEqualWithAccuracy(green, 0.0/255.0, 0.05,  @"Expected red value of 0, but got %f", green);
+    XCTAssertEqualWithAccuracy(blue, 0.0/255.0, 0.05,  @"Expected red value of 0, but got %f", blue);
+    XCTAssertEqual(alpha, 1,  @"Expected red value of 1, but got %f", alpha);
     
-    color1 = [UIColor tc_colorWithHexString:@"F00"];
-    color2 = [UIColor colorWithRed:1.0 green:0 blue:0 alpha:1.0];
+    UIColor *otherColor  = [UIColor tc_colorWithHexString:@"#E34"];
     
-    [color1 getRed:&r1 green:&g1 blue:&b1 alpha:&a1];
-    [color2 getRed:&r2 green:&g2 blue:&b2 alpha:&a2];
+    [otherColor getRed:&red green:&green blue:&blue alpha:&alpha];
     
-    XCTAssertTrue(r1-r2 == 0 && g1-g2 == 0 && b1-b2 == 0 && a1-a2 == 0);
+    XCTAssertEqualWithAccuracy(red, 238.0/255.0, 0.05,  @"Expected red value of %f, but got %f", 238.0/255.0, red);
+    XCTAssertEqualWithAccuracy(green, 51.0/255.0, 0.05,  @"Expected red value of %f, but got %f", 51.0/255.0, green);
+    XCTAssertEqualWithAccuracy(blue, 68.0/255.0, 0.05,  @"Expected red value of %f, but got %f", 68.0/255.0, blue);
+    XCTAssertEqual(alpha, 1,  @"Expected red value of 1, but got %f", alpha);
     
-    color2 = [UIColor blueColor];
-    [color2 getRed:&r2 green:&g2 blue:&b2 alpha:&a2];
-    
-    XCTAssertFalse(r1-r2 == 0 && g1-g2 == 0 && b1-b2 == 0 && a1-a2 == 0);
-    
-    color1 = [UIColor tc_colorWithHexString:@"00F"];
-    color2 = [UIColor colorWithHue:240.0/360.0 saturation:1.0 brightness:1.0 alpha:1.0];
+    otherColor = [UIColor tc_colorWithHexString:@"C12345"];
     
-    [color1 getRed:&r1 green:&g1 blue:&b1 alpha:&a1];
-    [color2 getRed:&r2 green:&g2 blue:&b2 alpha:&a2];
+    [otherColor getRed:&red green:&green blue:&blue alpha:&alpha];
     
-    XCTAssertTrue(r1-r2 == 0 && g1-g2 == 0 && b1-b2 == 0 && a1-a2 == 0);
+    XCTAssertEqualWithAccuracy(red, 193.0/255.0, 0.05,  @"Expected red value of %f, but got %f", 193.0/255.0, red);
+    XCTAssertEqualWithAccuracy(green, 35.0/255.0, 0.05,  @"Expected red value of %f, but got %f", 35.0/255.0, green);
+    XCTAssertEqualWithAccuracy(blue, 69.0/255.0, 0.05,  @"Expected red value of %f, but got %f", 69.0/255.0, blue);
+    XCTAssertEqual(alpha, 1,  @"Expected red value of 1, but got %f", alpha);
     
-    color2 = [UIColor redColor];
-    [color2 getRed:&r2 green:&g2 blue:&b2 alpha:&a2];
-    
-    XCTAssertFalse(r1-r2 == 0 && g1-g2 == 0 && b1-b2 == 0 && a1-a2 == 0);
+    XCTAssertThrows([UIColor tc_colorWithHexString:@""]);
+    XCTAssertThrows([UIColor tc_colorWithHexString:@"1"]);
+    XCTAssertThrows([UIColor tc_colorWithHexString:@"F"]);
+    XCTAssertThrows([UIColor tc_colorWithHexString:@"00"]);
+    XCTAssertThrows([UIColor tc_colorWithHexString:@"F000"]);
+    XCTAssertThrows([UIColor tc_colorWithHexString:@"C123456"]);
+    XCTAssertThrows([UIColor tc_colorWithHexString:@"C123456789"]);
```

Comment:
None of these tests try non hex characters which should be invalid.

For example, this test would fail:
```
XCTAssertThrows([UIColor tc_colorWithHexString:@"orange"]);
```
But it should pass.  Besides `a` and `e` those aren't valid hex characters.